app.rb

```diff
@@ -45,9 +50,15 @@ def handle_event(event:, context:)
 end
 
 def fetch_locations_and_respond(params)
-    records = $locations_manager.fetch_locations params['location_codes']
-rescue LocationsManagerError
-    $logger.info 'Received LocationsManager error'
+  req_codes = params['location_codes'].split(",")
+  records = req_codes.map do |req_code|
+    [
+      req_code,
+      $locations.select {|k,v| k.match? req_code}.map {|k,v| v}
```

Comment:
Small thing but I think this could be simplified with [fetch](https://ruby-doc.org/core-2.5.1/Hash.html#method-i-fetch) and just have it return `nil` if the key isn't found

Reply:
Does `fetch` work with regular expressions/multiple possible values? I don't know if multiple matches is ever something we need to support, but it does seem like the locations data we have is set up to allow multiple locations codes to all map to the same url e.g. possible lots of things match `ma.*`. I don't know if multiple matches is ever something we need to support

Reply:
I'm not sure either, but its not a bad idea to support it. If we do, obviously leave this as is, but also we should add some documentation to the readme that explains how this works otherwise I think that functionality would be quickly forgotten